src/load-styles.js

```diff
@@ -0,0 +1,89 @@
+import { maxSatisfying } from 'es-semver';
+
+const CDN_HOST = 'https://components.clever-cloud.com';
+const CSS = 'application/javascript';
+const ONE_YEAR = 365 * 24 * 60 * 60;
+
+export async function handleLoadStylesRequest (httpApi) {
+
+  const versionsList = await httpApi.fetch(`versions-list.json`, CDN_HOST);
+  if (versionsList == null) {
+    return httpApi.sendResponse({ body: '', status: 500 });
+  }
+
+  const version = getVersion(versionsList, httpApi.getQueryParam('version'));
+
+  const depsManifest = await httpApi.fetch(`deps-manifest-${version.resolved}.json`, CDN_HOST);
+  if (depsManifest == null) {
+    return httpApi.sendResponse({ body: '', status: 500 });
+  }
+
+  const response = await getResponse(version, depsManifest, httpApi);
+
+  return httpApi.sendResponse(response);
+}
+
+// Resolve latest possible version from versions list
+export function getVersion (versionsList, requestedVersion) {
+  // This code allows all the possible semver specifiers
+  // We'll only test and document the short and simple use cases
+  const resolvedVersion = maxSatisfying(versionsList, requestedVersion || '*');
+  return {
+    requested: requestedVersion,
+    resolved: resolvedVersion,
+  };
+}
+
+function getHeaders ({ type, maxAge }) {
+  return {
+    'content-type': type,
+    'access-control-allow-origin': '*',
+    'cache-control': `public,max-age=${maxAge}`,
+  };
+}
+
+export async function getResponse (version, depsManifest, httpApi) {
+
+  if (version.resolved == null) {
```

Comment:
this will never happen since the httpApi.fetch(``deps-manifest-${version.resolved}.json``, CDN_HOST) at line 14 may have failed

Reply:
You're 100% right, I guess I actually should fix #1 in this PR.

Reply:
Done